src/components/AddItemForm/AddItemForm.js

```diff
@@ -19,8 +28,55 @@ const AddItemForm = ({ listId }) => {
   const handleSubmit = async (event) => {
     event.preventDefault();
 
+    // check if itemName already exists in Firestore
+    try {
+      // get array of listItems from Firestore
+      await db
+        .collection(`lists/${listId}/items`)
+        .get()
+        .then(async (querySnapshot) => {
+          // create array (dbItemArray) by returning itemNames from Firestore response (querySnapshot)
+          let dbItemArray = querySnapshot.docs.map((doc) => {
+            return doc.data().itemName;
+          });
+
+          const duplicateResult = isItemDuplicate(
+            formValues.itemName,
+            dbItemArray,
+          );
+
+          // if item exists, show error message, otherwise, continue with adding to database
+          if (duplicateResult === true) {
+            setErrorMessage('Item already exists in Shopping List');
```

Comment:
For accessibility, it would be nice to focus the user on the text field after submission if there is an error, to save having to navigate to the invalid field. (We did not do this in the other PR's form either, so I will make a suggestion there as well.)